### DISCUSSION TITLE
Complete Vaccine Registration flow

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#421] [Ideas]
Add view and funtionality to
- [ ] Enrol a patient in a Covid-19 vaccine care plan and record-related vaccinations 
- [ ] Capture the required information to generate a vaccine certificate for the international patient summary. 
- [ ] Easily author / adapt the vaccination form using SDC/FHIR questionnaires. This form when completed will be programmed to automatically update the appropriate FHIR resources (eg. Immunization). A form builder will not be included in the work but these forms can easily be authored with a tool like (https://lhcformbuilder.nlm.nih.gov) 
- [ ] Generate a Care Plan  - When the patient has registered a care plan for Covid-19 will be set-up with tasks created for a single or multi-dose administration (depending on the vaccine being administered).

**Comment by rkodev** [comment]
A conceptual UI Flow Diagram of the Covax App
![Covax Flow 2](https://user-images.githubusercontent.com/43806892/115566224-be993980-a2c2-11eb-9864-aa504771c826.jpg)

**Comment by pld** [comment]
Cool, this size looks like a story, I think we should break this down into smaller issues for assignment.

That flow looks ok to me CC @isabelcshaw @rowo

**Comment by rkodev** [comment]
@pld Agreed, we will need to break this issue to smaller issues. I've also updated the diagram to include vaccine certificate

**Comment by rkodev** [comment]
Covax App represented using FHIR Resources.

![Covax FHIR Modeling](https://user-images.githubusercontent.com/43806892/115570156-8eec3080-a2c6-11eb-8758-e87e42c8d7f0.jpg)

The records in `blue` are FHIR resources

@ekigamba and I modeled the Vaccine Certificate using a Medication Statement, the QR code can be represented using the `MedicationStatement.identifier` property.

The diagram still needs more research in
1. Generating the Care Plan, atm recording the selected Care Plan in a Medication Statement can be used to evaluate the Care Plan
2. Task Generation. Generating the next task that is dependent on the previous task completion e.g recording a second dose. This can be done by either generating all the tasks during Care Plan Generation or by reevaluation of the care plan after a task is completed. This needs more research

cc @ekigamba
